File: katas.py
# Python Kata Training
# Codewars Challenges
# author: junowoz

#################################################################################################################################################################
#1
#Timmy & Sarah think they are in love, but around where they live, they will only know once they pick a flower each. 
#If one of the flowers has an even number of petals and the other has an odd number of petals it means they are in love.
#Write a function that will take the number of petals of each flower and return true if they are in love and false if they aren't.

def lovefunc( flower1, flower2 ):

    if (flower1 % 2 == 0) and (flower2 % 2 != 0):
         return True
    elif (flower1 % 2 != 0) and (flower2 % 2 == 0):
        return True
    else :
        return False
    
    
lovefunc(5, 5);

#################################################################################################################################################################
#2        
#Your task is to write a function which returns the time since midnight in milliseconds.
#Example:
#h = 0
#m = 1
#s = 1
#result = 61000  

def past(h, m, s):
    # Good Luck!
    return h * 3600000 + m * 60000 + s * 1000

past(0,1,1)

#################################################################################################################################################################
#3
#Create a function which answers the question "Are you playing banjo?".
#If your name starts with the letter "R" or lower case "r", you are playing banjo!

def are_you_playing_banjo(name):
    if (name.startswith("R") or name.startswith("r")):
        return name + " plays banjo" 
    else:
        return name + " does not play banjo"

#################################################################################################################################################################
#4
#Complete the function so that it finds the average of the three scores passed to it and returns the letter value associated with that grade.

def get_grade(s1, s2, s3):
    # Code here
    score = (s1+s2+s3)/3
    if (90 <= score <= 100):
        return 'A'
    elif (80 <= score < 90):
        return 'B'
    elif (70 <= score < 80):
        return 'C'
    elif (60 <= score < 70):
        return 'D'
    elif (0 <= score < 60):
        return 'F'

#################################################################################################################################################################
#5
#We need a function that can transform a string into a number. What ways of achieving this do you know?
#Note: Don't worry, all inputs will be strings, and every string is a perfectly valid representation of an integral number.
def string_to_number(s):
    # your code here
    return int(s)

#################################################################################################################################################################
#6
#Write a function which calculates the average of the numbers in a given list.
#Note: Empty arrays should return 0.
def find_average(numbers):
    return sum(numbers) / len(numbers)
         
#################################################################################################################################################################
#7 Beginner Series #4 Cockroach
import math
def cockroach_speed(s):
    return math.floor(s * 100000 / 3600);

#################################################################################################################################################################
#8

File: katas.js
/**
 * JavaScript Kata Training
 * Codewars Challenges
 * @author: junowoz
 */

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 1
 * Complete the solution so that it reverses the string passed into it.
 *  'world'  =>  'dlrow'
 *  'word'   =>  'drow'
 */

function solution(str){  
    return str.split("").reverse().join("");
}

solution("world")
solution('word')


// i will be an integer. Double it and return it.
function doubleInteger(i) {
  i *= 2;
  return i;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 2
 * We need a function that can transform a number into a string.
 * What ways of achieving this do you know?
 * Examples:
 * 123 --> "123"
 * 999 --> "999"
 */
function numberToString(num) {
  // Return a string of the number here!
  return num.toString()
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 3
 * 
 * There was a test in your class and you passed it. Congratulations!
 * But you're an ambitious person. You want to know if you're better than the average student in your class.
 * You receive an array with your peers' test scores. Now calculate the average and compare your score!
 * Return True if you're better, else False!
 */

function betterThanAverage(classPoints, yourPoints) {
  // Your code here
  let average = 0
  
  for (let i in classPoints){
    average += (classPoints[i]) / (classPoints.length)
  }
  
  if (yourPoints > average){
    return true
  } else {
    return false
  }
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 4
 * Your classmates asked you to copy some paperwork for them. You know that there are 'n' classmates and the paperwork has 'm' pages.
 * Your task is to calculate how many blank pages do you need. If n < 0 or m < 0 return 0.
 * n= 5, m=5: 25
 */

function paperwork(n, m) {
  if (n < 0 || m < 0)
    return 0
  else
    return n* m
  
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 5
 * Implement a function which multiplies two numbers.
 */
//The shortest way I found
const multiply = (a, b) => a * b 

//The long way
function multiply(a, b) {
  return a * b
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 6
 * Create a function that gives a personalized greeting. This function takes two parameters: name and owner.
 * Use conditionals to return the proper message:
 */

function greet (name, owner) {
  // Add code here
      if (name == owner)
        return 'Hello boss'
      else
        return 'Hello guest'
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/** 7
 * Sure, this is about as easy as it gets. 
 * But how clever can you be to create the most creative hello world you can think of?
 * What is a "hello world" solution you would want to show your friends?
 */
// Write a function "greet" that returns "hello world!"
const greet = greet => 'hello world!'

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 8
// Write a function that takes an array of words and smashes them together into a sentence and returns the sentence.
// You can ignore any need to sanitize words or add punctuation, but you should add spaces between each word.

function smash(words) {
  return words.join(" ");
}


/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 9
/*
Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.
The output should be two capital letters with a dot separating them.
It should look like this:
Sam Harris => S.H
patrick feeney => P.F
 */
function abbrevName(name){
  const nameArray = name.split(" ");
  return (nameArray[0][0] + "." + nameArray[1][0]).toUpperCase();
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 10
/*
Simple, remove the spaces from the string, then return the resultant string.
 */
function noSpace(x){
  return x.replace(/ /g, '')
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/*
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
You may assume that each input would have exactly one solution, and you may not use the same element twice.
You can return the answer in any order.
 */
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  for (let index = 0; index < nums.length; index++) {
    const diff = target - nums[index];
    const diffIndex = nums.indexOf(diff);
    // "diffIndex !== index" takes care of same index not being reused
    if (diffIndex !== -1 && diffIndex !== index) {
      return [index, diffIndex];
    }
  }
};

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 11
/*
Your task is to convert strings to how they would be written by Jaden Smith. 
The strings are actual quotes from Jaden Smith, but they are not capitalized in the same way he originally typed them. */

String.prototype.toJadenCase = function () {
    var returnString = [];
    var words = this.toLowerCase().split(' ');
    
    for (var i = 0; i < words.length; i++) {
        word = words[i];
        returnString.push(word[0].toUpperCase() + word.slice(1));
    }
    return returnString.join(" ");
};